src/Microsoft.Health.Checkpoints/Storage/BaseStorageCheckpointClient.cs

```diff
@@ -0,0 +1,90 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using EnsureThat;
+
+namespace Microsoft.Health.Checkpoints.Storage
+{
+    public class BaseStorageCheckpointClient : ICheckpointClient
+    {
+        private BlobContainerClient _storageClient;
+
+        public BaseStorageCheckpointClient(BlobContainerClient containerClient)
+        {
+            EnsureArg.IsNotNull(containerClient);
+
+            _storageClient = containerClient;
+        }
+
+        public virtual Task<ICheckpoint> GetCheckpointAsync(string checkpointIdentifier)
+        {
+            Task<ICheckpoint> GetCheckpointAsync()
+            {
+                ICheckpoint checkpoint = new Checkpoint();
+
+                foreach (BlobItem blob in _storageClient.GetBlobs(traits: BlobTraits.Metadata, states: BlobStates.All, prefix: checkpointIdentifier, cancellationToken: CancellationToken.None))
+                {
+                    DateTimeOffset lastEventTimestamp = DateTime.MinValue;
+
+                    if (blob.Metadata.TryGetValue("LastProcessedDt", out var dt))
```

Comment:
Should this be an error condition? If the checkpoint is empty or partially empty is it still valid?  Would at a minimum suggest injecting an ILogger and log a warning.

Reply:
nit: `LastProcessedDt` is used several times and could be a const

Reply:
Updated to log a warning and now use a const